From e98ce8acf84d12ea67a3ac76bf63c6d87d9af86d Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Mon, 20 May 2013 16:38:06 +0200
Subject: [PATCH] test: fix insufficient quoting of '\'

This is a follow-up to 7f2c91b8369242a8dbc2b304a5b71b2a85f5b855, which
caused sbits-restore.test to fail in the following way in case SELinux
was disabled:

*** sbits-restore.test ***
[3] $ umask 022 -- ok
[4] $ mkdir d -- ok
[5] $ touch d/g -- ok
[6] $ touch d/u -- ok
[7] $ chmod u+s d/u -- ok
[8] $ chmod g+s d/g -- ok
[9] $ chmod +t d -- ok
[10] $ getfacl -R d > d.acl -- ok
[11] $ rm -R d -- ok
[12] $ mkdir d -- ok
[13] $ touch d/g -- ok
[14] $ touch d/u -- ok
[15] $ setfacl --restore d.acl -- ok
[16] $ ls -dl d | awk '{print $1}' | sed 's/.$//g' -- failed
drwxr-xr-                             != drwxr-xr-t
[18] $ ls -dl d/u | awk '{print $1}' | sed 's/.$//g' -- failed
-rwSr--r-                             != -rwSr--r--
[20] $ ls -dl d/g | awk '{print $1}' | sed 's/.$//g' -- failed
-rw-r-Sr-                             != -rw-r-Sr--
[22] $ rm -Rf d -- ok
17 commands (14 passed, 3 failed)

Upstream-Status: Backport
http://git.savannah.gnu.org/cgit/acl.git/commit/?id=e98ce8acf84d12ea67a3ac76bf63c6d87d9af86d

Signed-off-by: Kamil Dudka <kdudka@redhat.com>
Signed-off-by: He Zhe <zhe.he@windriver.com>
---
 test/cp.test            | 2 +-
 test/misc.test          | 6 +++---
 test/sbits-restore.test | 6 +++---
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/test/cp.test b/test/cp.test
index 0867f63..a927195 100644
--- a/test/cp.test
+++ b/test/cp.test
@@ -9,7 +9,7 @@ The cp utility should only copy ACLs if `-p' is given.
 	> -rw-rw-r--+

 	$ cp f g
-	$ ls -l g | awk -- '{ print $1 }' | sed 's/\.$//g'
+	$ ls -l g | awk -- '{ print $1 }' | sed 's/\\.$//g'
 	> -rw-r--r--

 	$ rm g
diff --git a/test/misc.test b/test/misc.test
index 6e98053..29372b7 100644
--- a/test/misc.test
+++ b/test/misc.test
@@ -254,7 +254,7 @@ Add some users and groups
 Symlink in directory with default ACL?

 	$ ln -s d d/l
-	$ ls -dl d/l | awk '{print $1}' | sed 's/\.$//g'
+	$ ls -dl d/l | awk '{print $1}' | sed 's/\\.$//g'
 	> lrwxrwxrwx

 	$ ls -dl -L d/l | awk '{print $1}'
@@ -343,7 +343,7 @@ Remove the default ACL
 Reset to base entries

 	$ setfacl -b d
-	$ ls -dl d | awk '{print $1}' | sed 's/\.$//g'
+	$ ls -dl d | awk '{print $1}' | sed 's/\\.$//g'
 	> drwxr-x---

 	$ getfacl --omit-header d
@@ -355,7 +355,7 @@ Reset to base entries
 Now, chmod should change the group_obj entry

 	$ chmod 775 d
-	$ ls -dl d | awk '{print $1}' | sed 's/\.$//g'
+	$ ls -dl d | awk '{print $1}' | sed 's/\\.$//g'
 	> drwxrwxr-x

 	$ getfacl --omit-header d
diff --git a/test/sbits-restore.test b/test/sbits-restore.test
index de21340..5899b0a 100644
--- a/test/sbits-restore.test
+++ b/test/sbits-restore.test
@@ -13,10 +13,10 @@ Ensure setting of SUID/SGID/sticky via --restore works
 	$ touch d/g
 	$ touch d/u
 	$ setfacl --restore d.acl
-	$ ls -dl d | awk '{print $1}' | sed 's/\.$//g'
+	$ ls -dl d | awk '{print $1}' | sed 's/\\.$//g'
 	> drwxr-xr-t
-	$ ls -dl d/u | awk '{print $1}' | sed 's/\.$//g'
+	$ ls -dl d/u | awk '{print $1}' | sed 's/\\.$//g'
 	> -rwSr--r--
-	$ ls -dl d/g | awk '{print $1}' | sed 's/\.$//g'
+	$ ls -dl d/g | awk '{print $1}' | sed 's/\\.$//g'
 	> -rw-r-Sr--
 	$ rm -Rf d
--
1.9.1

